src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsProjectJsonToPackageReferenceMigrateResult.cs

```
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.PackageManagement.VisualStudio;
+
+namespace NuGet.VisualStudio
+{
+    internal class VsProjectJsonToPackageReferenceMigrateResult : IVsProjectJsonToPackageReferenceMigrateResult
+    {
+        public bool IsSuccess { get; }
+
+        public string ErrorMessage { get; }
+
+        public VsProjectJsonToPackageReferenceMigrateResult(ProjectJsonToPackageRefMigrateResult result)
```

why can't have single constructor with both values with default?

---

one is created when we actually have a projectjsontopackagerefmigrateresult object when the whole operation is successful. the other is created when an exception is thrown and we need to return a failure via this object.

---

i get that, but at the end it only needs a bool and a string which can easily be done with a single constructor for both cases